l10n_th_personal_income_tax/models/pit_move.py

```diff
@@ -0,0 +1,55 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+from odoo.addons.l10n_th_withholding_tax_cert.models.withholding_tax_cert import (
+    WHT_CERT_INCOME_TYPE,
+)
+
+
+class PersonalIncomeTaxMove(models.Model):
+    _name = "pit.move"
+    _description = "Personal Income Tax Move"
+
+    payment_id = fields.Many2one(
+        comodel_name="account.payment",
+        string="Payment",
+        index=True,
+        required=True,
+        ondelete="cascade",
+        domain=[("state", "not in", ["draft", "cancel"])],
+    )
+    payment_state = fields.Selection(related="payment_id.state")
+    partner_id = fields.Many2one(
+        comodel_name="res.partner",
+        string="Vendor",
+        index=True,
+        required=True,
+        ondelete="cascade",
+    )
+    cancelled = fields.Boolean(readonly=True, help="for filtered Payment is cancel")
+    date = fields.Date(
+        compute="_compute_date",
+        store=True,
+    )
+    calendar_year = fields.Char(
+        string="Calendar Year",
+        compute="_compute_date",
+        store=True,
```

Comment:
I think calendar_year can be pivotal field for report.
```suggestion
        store=True,
        index=True,
```